website/docs/docs/collaborate/govern/model-versions.md

```diff
@@ -9,32 +9,76 @@ description: "Version models to help with lifecycle management"
 
 Model versions are different. Multiple versions of a model will live in the same code repository at the same time and be deployed into the same data environment simultaneously. This is similar to how web APIs are versioned—multiple versions are live simultaneously; older versions are often eventually sunsetted.
 
-dbt's model `versions` makes it possible to define multiple versions:
-- That share the same "reference" name
-- While reusing the same top-level properties, highlighting just their differences
+## How is this different from just creating a new model?
+
+Honestly, it's only a little bit different! There isn't much magic here, and that's by design.
+
+You've always been able to create a new model, and name it `dim_customers_v2`. Why should you opt for a "real" versioned model instead?
+
+First, the versioned model preserves its _reference name_. Versioned models are `ref`'d by their _model name_, rather than the name of the file that they're defined in. By default, the `ref` resolves to the latest version (as declared by that model's maintainer), but you can also `ref` a specific version of the model, with a `version` keyword.
+
+<File name="models/schema.yml">
+
+```sql
+{{ ref('dim_customers') }}            -- resolves to latest
+{{ ref('dim_customers', version=2) }} -- resolves to v2
+```
+
+</File>
+
+Second, a versioned model can reuse the majority of its yaml properties and configuration. Each version needs to only say how it _differs_ from the shared set of attributes. This gives you an opportunity to highlight the differences across versions—which is otherwise difficult to detect in models with dozens or hundreds of columns—and to clearly track, in one place, all versions of the model which are currently live.
+
+Third, dbt supports `version`-based selection. For example, you could define a [default yaml selector](node-selection/yaml-selectors#default), to avoid running any old model versions in development—even as you continue to run them in production through a sunset and migration period:
+
+```yml
+selectors:
+  - name: exclude_old_versions
+    default: "{{ target.name == 'dev' }}"
+    definition:
+      method: fqn
+      value: "*"
+      exclude:
+        - method: version
+          value: old
+```
+
+Finally, we intend to add support for **deprecating models** in dbt Core v1.6. When you slate a versioned model for deprecation, dbt will be able to provide more helpful warnings to downstream consumers of that model. Rather than just, "This model is going away," it's - "This older version of the model is going away, and there's a new version coming soon."
```

Comment:
>"This older version of the model is going away, and there's a new version coming soon."

I read this as "the new version hasn't been developed yet" - do you mean people might deprecate models ahead of when their replacements have been deployed "behind the feature flag" of not being `latest`?

Do we give a warning if `latest` is v3 and I query v2 by name, even if deprecation isn't scheduled? If so then it's more like 
>Rather than just, "This model has been superseded," it's - "This model has been superseded, and you need to act."

Reply:
Good point - my wording here was not clear.

We'll support `deprecation_date` for both versioned and non-versioned models. I was trying to differentiate here between the experience of downstream consumers, when an upstream producer deprecates a model that's versioned, versus a standalone (non-versioned) model. Again, trying to make the case for - why use versions at all? Why not just create two separate SQL files, `my_model_old` and `my_model_new`, and dbt is none the wiser?

> Do we give a warning if `latest` is v3 and I query v2 by name, even if deprecation isn't scheduled?

As a **consumer**, I don't think we'd give a warning if you query `v2` by name, and it's not scheduled for deprecation. I think we'd strongly encourage **producers** to put a deprecation date on `v2` at the same time that they roll over `latest_version: 3`.

At first, I thought you were asking something slightly different - should we warn the **consumer** about an unpinned reference to a model, as soon as a prerelease version becomes available?
```
12:53:34  4 of 4 START sql view model dbt_jcohen.another_model ........................... [RUN]
12:53:34  FYI!
Found an unpinned reference to versioned model 'my_model' in 'my_dbt_project'.
Resolving to latest version: my_model.v2
A prerelease version v3 is available. It has not yet been marked 'latest' by its maintainer.
When that happens, this reference will resolve to my_model.v3 instead.

  Try out v3: {{ ref(my_dbt_project, my_model, v=3) }}
  Pin to v2:  {{ ref(my_dbt_project, my_model, v=2) }}
```